src/ui/public/courier/saved_object/saved_object.js

```diff
@@ -231,59 +240,54 @@ export default function SavedObjectFactory(es, kbnIndex, Promise, Private, Notif
     };
 
     /**
-     * Save this object
-     *
-     * @return {Promise}
-     * @resolved {String} - The id of the doc
+     * Checks the id of the instance. If it is not set, creates a uuid for it.
      */
-    self.save = function () {
-
-      let body = self.serialize();
-
-      // Slugify the object id
-      self.id = slugifyId(self.id);
+    self.ensureId = function () {
```

Comment:
Does this need to be publicly accessible? If not, we can just declare it within the closure:

```javascript
function ensureId() {
  // ...
}
```

We could even move it into the `save` method.